src/main/webapp/scripts/script.js

```diff
@@ -76,46 +77,97 @@ function toggleLogin() {
-    alert("Please enter a valid email address.");
+    alert('Please enter a valid email address.');
   }
   if (password == null || password.length < 2) {
-    alert("Please enter a valid password.");
+    alert('Please enter a valid password.');
   }
-  
+
   // If email/pass are of appropriate length, try to create a new account.
-  firebase.auth().createUserWithEmailAndPassword(email, password)
+  firebase.auth()
+      .createUserWithEmailAndPassword(email, password)
       .catch(function onFailure(error) {
         var errorCode = error.code;
         var errorMessage = error.message;
 
         if (errorCode == 'auth/weak-password') {
-          alert('The password is too weak. Passwords must be at least 6 characters long.');
+          alert(
+              'The password is too weak. Passwords must be at least 6 characters long.');
         } else if (errorCode == 'auth/invalid-email') {
           alert('Please enter a valid email address.');
         } else if (errorCode == 'auth/email-already-in-use') {
           alert('This email address is already associated with a user.');
-        } else { alert(errorMessage); }
-  });
+        } else {
+          alert(errorMessage);
+        }
+      });
 }
 
 function sendPasswordResetEmail() {
-  var email = document.getElementById("email").value;
-    
-  firebase.auth().sendPasswordResetEmail(email).then(function onSuccess() {
-    alert('Password Reset Email Sent!');
-    }).catch(function(error) {
-      // Handle errors.
-      var errorCode = error.code;
-      var errorMessage = error.message;
-    
-      if (errorCode == 'auth/invalid-email') {
-        alert("Please enter a valid email address.");
-      } else if (errorCode == 'auth/user-not-found') {
-        alert("This user was not found.");
-      } else { alert(errorMessage); }
+  var email = document.getElementById('email').value;
+
+  firebase.auth()
+      .sendPasswordResetEmail(email)
+      .then(function onSuccess() {
+        alert('Password Reset Email Sent!');
+      })
+      .catch(function(error) {
+        // Handle errors.
+        var errorCode = error.code;
+        var errorMessage = error.message;
+
+        if (errorCode == 'auth/invalid-email') {
+          alert('Please enter a valid email address.');
+        } else if (errorCode == 'auth/user-not-found') {
+          alert('This user was not found.');
+        } else {
+          alert(errorMessage);
+        }
+      });
+}
+
+function getRecipeList() {  // eslint-disable-line no-unused-vars
+  fetch('/api/post').then((response) => response.json()).then((recipes) => {
+    const recipeEl = document.getElementById('recipe-list');
+    removeAllChildNodes(recipeEl);
+    recipes.forEach((recipe) => {
+      recipeEl.appendChild(createRecipeListElement(recipe));
     });
-}
\ No newline at end of file
+  });
+}
+
+function getDetailedRecipe() {  // eslint-disable-line no-unused-vars
+  fetch('/api/post?recipeID=zElvE9jKb1BX2CDN608O')
```

Comment:
Not sure why this has a hard coded ID here instead of taking a recipe ID as input.

Reply:
On it